src/core/geometry/qgsgeometry.cpp

```diff
@@ -606,6 +606,10 @@ double QgsGeometry::sqrDistToVertexAt( QgsPointXY &point, int atVertex ) const
 
 QgsGeometry QgsGeometry::nearestPoint( const QgsGeometry &other ) const
 {
+  if ( isEmpty() )
```

Comment:
Wouldn't it be better if this was handled in `QgsGeos::closestPoint` instead?

Reply:
Yes I agree. Done.